cmake/ConfigOptions.cmake

```diff
@@ -150,6 +150,11 @@ CMAKE_DEPENDENT_OPTION(WITH_VIDEO_FFMPEG "Use FFMPEG for video encoding/decoding
 	"WITH_FFMPEG" OFF)
 CMAKE_DEPENDENT_OPTION(WITH_VAAPI "Use FFMPEG VAAPI" OFF
 	"WITH_VIDEO_FFMPEG" OFF)
+CMAKE_DEPENDENT_OPTION(WITH_VAAPI_H264_ENCODING "Use FFMPEG VAAPI hardware H264 encoding" OFF
```

Comment:
It means `WITH_VAAPI_H264_ENCODING` is dependent on `WITH_VIDEO_FFMPEG` being enabled, because VAAPI H264 encoding is only supported by ffmpeg library (not by openh264).

Also, to answer your concern, seems like `WITH_VAAPI_H264_ENCODING` in addition to existing `WITH_VAAPI` provides more flexibility, because VAAPI can be enabled independently for H264 encoding and decoding.